es-core/src/components/ImageComponent.cpp

```diff
@@ -282,18 +276,18 @@ void ImageComponent::updateVertices()
 
 	// we go through this mess to make sure everything is properly rounded
 	// if we just round vertices at the end, edge cases occur near sizes of 0.5
-	const Vector2f     size        = { Math::round(mSize.x()), Math::round(mSize.y()) };
-	const Vector2f     topLeft     = { size * mTopLeftCrop };
-	const Vector2f     bottomRight = { size * mBottomRightCrop };
-	const float        px          = mTexture->isTiled() ? mSize.x() / getTextureSize().x() : 1.0f;
-	const float        py          = mTexture->isTiled() ? mSize.y() / getTextureSize().y() : 1.0f;
-	const unsigned int color       = Renderer::convertColor(mColorShift);
-	const unsigned int colorEnd    = Renderer::convertColor(mColorShiftEnd);
-
-	mVertices[0] = { { topLeft.x(),     topLeft.y()     }, { mTopLeftCrop.x(),          py   - mTopLeftCrop.y()     }, color };
-	mVertices[1] = { { topLeft.x(),     bottomRight.y() }, { mTopLeftCrop.x(),          1.0f - mBottomRightCrop.y() }, mColorGradientHorizontal ? colorEnd : color };
-	mVertices[2] = { { bottomRight.x(), topLeft.y()     }, { mBottomRightCrop.x() * px, py   - mTopLeftCrop.y()     }, mColorGradientHorizontal ? color : colorEnd };
-	mVertices[3] = { { bottomRight.x(), bottomRight.y() }, { mBottomRightCrop.x() * px, 1.0f - mBottomRightCrop.y() }, color };
+	const Vector2f size        = { Math::round(mSize.x()), Math::round(mSize.y()) };
+	const Vector2f topLeft     = { size * mTopLeftCrop };
+	const Vector2f bottomRight = { size * mBottomRightCrop };
+	const float    px          = mTexture->isTiled() ? mSize.x() / getTextureSize().x() : 1.0f;
+	const float    py          = mTexture->isTiled() ? mSize.y() / getTextureSize().y() : 1.0f;
+
+	mVertices[0] = { { topLeft.x(),     topLeft.y()     }, { mTopLeftCrop.x(),          py   - mTopLeftCrop.y()     }, 0xFFFFFFFF };
```

Comment:
Want to get feedback on this change.  Trying to avoid duplication of logic from `updateColors` here.  For now, I am just setting color to 0xFFFFFFFF and then calling updateColors after.

Reply:
updateVertices is called by resize & setFlipX/Y. When these are called, color don't change.

The current problem is related to the fact that the assignment is done by allocating a complete new instance of Vertex and assign it by copy, including color.

It may be simpler to split assignent of vertex, and avoid changing colors like that.

mVertices[0].pos = { topLeft.x(),     topLeft.y()     };
mVertices[0].text = { mTopLeftCrop.x(),          py   - mTopLeftCrop.y()     }

Reply:
Going to leave assignment as is.  `updateVertices` shouldn't be called much (in most cases, it should only be called once per ImageComponent), so I don't think allocating a new instance is an issue.  Switched initial color assignment to `0` for consistency, since this is what NinePatchComponent was already doing.